com.unity.multiplayer.mlapi/Runtime/Messaging/CustomMessageManager.cs

```diff
@@ -55,7 +55,20 @@ public void SendUnnamedMessage(List<ulong> clientIds, NetworkBuffer buffer, Netw
                 return;
             }
 
-            m_NetworkManager.MessageSender.Send(NetworkConstants.UNNAMED_MESSAGE, networkChannel, clientIds, buffer);
+            var context = m_NetworkManager.MessageQueueContainer.EnterInternalCommandContext(
+                MessageQueueContainer.MessageType.UnnamedMessage,
+                networkChannel,
+                clientIds.ToArray(),
```

Comment:
Additional allocations from `ToArray` which were not present with the old custom messaging? Have we considered using lists instead of arrays for the command context? It looks like we allocate here for every command we send?

Reply:
Same question as above... is there any reason a stackalloc array wouldn't work here?

Reply:
I'm not sure if we can stackalloc this because the resulting value will need to persist in the message queue (maybe I'm wrong about this?).

So in an ideal world we would want the following here:
- pool the data structure which holds client ids.
- use a data structure which is resizable (arraysegment, list, nativearraysegment, nativelist)

I raised this with @MFatihMAR and @NoelStephensUnity when RPC queue was implemented but we did not address this back then. Maybe we should reconsider cleaning this up now as the problem is spreading into other feature areas causing allocations on every single command we send.

Reply:
yeah, I also dislike `.ToArray()` there. there are a few things involved here: LINQ-like RPC params extensions, internals of NetworkBehaviour.BeginXXXRpc/EndXXXRpc and internals of MessageQueue (formerly RpcQueue). we have some items in our tech debt list spreadsheet and feels like once we start killing some of these issues, this would be one of them. I'm just not sure when/how (thinking out loud here). perhaps add this ("it is also plaguing other features") as a supportive argument under RPC params, Collections etc?

Reply:
Actually `stackalloc` should work here. The array itself isn't persisted, it's written to a buffer. It is stored on the InternalCommandContext struct, but that struct itself is not intended to persist beyond the calling method, so it should be fine to pass stack allocated memory to it.

That said, the RFC I'm working on right now is partially about eliminating allocations in our message-sending code and is going to rework a lot of this (EnterInternalCommandContext itself is probably going to be short-lived, assuming my other proposal is accepted) so I'm not sure if it's worth going back and doing this work now or if it's better to just pick that up with the next task's changes.

Reply:
personally, I think we should leave alloc optimization for later. I can see some other plans and feats playing some role here too (hence my previous comment). I wouldn't bother too much under this PR.

Reply:
As mentioned in a previous comment, I've decided to make these part of my current WIP serialization RFC instead of addressing them now.